### PR TITLE
Clustering in Unsupervised Machine Learning

### LINKED ISSUE #625: Clustering in Unsupervised Machine Learning
### Brief Summary: 
Unsupervised learning is a technique in machine learning (ML) that does not require supervision of models by users. This article introduces readers to unsupervised machine learning and why it is needed. It will mainly focus on advancing knowledge in clustering and various clustering types in unsupervised ML.

### Key Takeaways: 
The following are key concepts that readers should understand;
1. Introduction to unsupervised machine learning
2. Why Unsupervised Learning is important
3. Clustering Analysis
4. Types of Clustering in Unsupervised ML (including algorithms, advantages and disadvantages of each type)
5. Conclusion

### References: 
https://www.datarobot.com/wiki/unsupervised-machine-learning/

### PR DESCRIPTION
**Pull request by mbaabuones**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      

#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [ ] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

This fixes #625 (Clustering in Unsupervised Machine Learning). I have uploaded seven images.

### CODE REVIEW

**Comment by lalith1403** [comment]
@mbaabuones before we start reviewing this article, I observed a few additional files￼. Can you ensure that we have files only related to this article￼?

**Comment by mbaabuones** [comment]
@lalith1403 there are seven image files related to the article 'Clustering in Unsupervised Machine Learning'-(cluster inertia, dbscan clustering, gaussian mixture model, hierarchical clustering dendrogram, illustration of clustering, k-mean algorithm and squared euclidean distance.

**Comment by lalith1403** [comment]
There is an article on auto scaling in aws as well! Go to files changed and you will find the same.

**Comment by lalith1403** [comment]
image

Check this @mbaabuones

File: articles/clustering-in-unsupervised-ml/index.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
+## Clustering in Unsupervised Machine Learning
+
+### Introduction to Unsupervised Machine Learning
+Unsupervised learning is a technique in machine learning (ML) that does not require the supervision of models by users. It is one of the categories of machine learning. The other two categories include reinforcement and supervised learning.
```
Comment (lalith1403) [inline articles/clustering-in-unsupervised-ml/index.md:4]: is a machine learning technique...

File: articles/clustering-in-unsupervised-ml/index.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
+## Clustering in Unsupervised Machine Learning
+
+### Introduction to Unsupervised Machine Learning
+Unsupervised learning is a technique in machine learning (ML) that does not require the supervision of models by users. It is one of the categories of machine learning. The other two categories include reinforcement and supervised learning. 
+
+In unsupervised machine learning, a self-organized algorithm is utilized to enhance the self-discovery of unknown patterns that are in unlabeled datasets.
```
Comment (lalith1403) [inline articles/clustering-in-unsupervised-ml/index.md:6]: In unsupervised machine learning, a self-organized algorithm is used to enhance the self-discovery of unknown patterns in unlabeled datasets.

File: articles/clustering-in-unsupervised-ml/index.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+### Introduction to Unsupervised Machine Learning
+Unsupervised learning is a technique in machine learning (ML) that does not require the supervision of models by users. It is one of the categories of machine learning. The other two categories include reinforcement and supervised learning. 
+
+In unsupervised machine learning, a self-organized algorithm is utilized to enhance the self-discovery of unknown patterns that are in unlabeled datasets. 
+
+This is contrary to supervised machine learning that uses human-labeled data. Unsupervised learning algorithms use unstructured data that is grouped based on similarities and patterns.
+
+### Why Unsupervised Learning is Important
+
+Unsupervised learning is an important concept in machine learning. It saves data analysts’ time by providing algorithms that enhance the grouping and investigation of data. It is also important in well-defined network models.  Many analysts prefer using unsupervised learning in network traffic analysis (NTA) because of frequent data changes and scarcity of labels.
```
Comment (lalith1403) [inline articles/clustering-in-unsupervised-ml/index.md:12]: Remove space after network models.

File: articles/clustering-in-unsupervised-ml/index.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+This is contrary to supervised machine learning that uses human-labeled data. Unsupervised learning algorithms use unstructured data that is grouped based on similarities and patterns.
+
+### Why Unsupervised Learning is Important
+
+Unsupervised learning is an important concept in machine learning. It saves data analysts’ time by providing algorithms that enhance the grouping and investigation of data. It is also important in well-defined network models.  Many analysts prefer using unsupervised learning in network traffic analysis (NTA) because of frequent data changes and scarcity of labels. 
+
+It is needed to create better forecasting, especially in threat detection. This can be achieved by developing network logs that enhance threat visibility. 
+
+This category of machine learning is also resourceful in the reduction of data dimensionality. Dimensionality reduction is needed in datasets that have very many features.Unsupervised learning can analyze complex data to establish less relevant features. The model can then be simplified by dropping these features with insignificant effects on valuable insights.
```
Comment (lalith1403) [inline articles/clustering-in-unsupervised-ml/index.md:16]: Space before Unsupervised learning can analyze...

File: articles/clustering-in-unsupervised-ml/index.md (hunk 2 above)
Comment (lalith1403) [reply articles/clustering-in-unsupervised-ml/index.md:6]: Can you rephrase this into a simpler sentence? It doesn't sound effortless. Also, what is a self-organized algorithm? Is self in self-discovery necessary?

File: articles/clustering-in-unsupervised-ml/index.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+
+###### Disadvantages
++ Hierarchical models have a high sensitivity to outliers. In the presence of outliers, the models do not perform well
++ The computation of hierarchical clustering is costly.
+
+##### Density-Based Spatial Clustering of Applications with Noise (DBSCAN)
+This is a density-based clustering that involves the grouping of data points that are close to each other. Data points that are far from each other are marked as outliers. Data is sorted based on commonalities. 
+
+###### Algorithm steps
+1.	In the first step, a core point should be identified. The core point radius is given as ε. Create a group  for each core point.
```
Comment (lalith1403) [inline articles/clustering-in-unsupervised-ml/index.md:102]: What is the core point? Define core point.

File: articles/clustering-in-unsupervised-ml/index.md [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+3.	Maximization Phase- The Gaussian parameters (mean and standard deviation) should be re-calculated using the ‘expectations’.
+4.	Evaluate whether there is convergence by examining the log-likelihood of existing data. 
+5.	Repeat steps 2-4 until there is convergence. 
+
+###### Advantages
++ It offers flexibility in terms of size and shape of clusters.
++ Membership can be assigned to multiple clusters, which makes it a fast algorithm for mixture models. 
+
+###### Disadvantages
++ If a mixture consists of insufficient points, the algorithm may diverge and establish solutions that contain infinite likelihood. This may require artificial regularization of covariance.
```
Comment (lalith1403) [inline articles/clustering-in-unsupervised-ml/index.md:143]: Artificial regularization of covariance? What does this mean, and what is its significance?

**Review by lalith1403 — CHANGES_REQUESTED** [review]
Amazing article. I learned a lot about unsupervised learning algorithms. It is well versed and well written. A few minor changes and this article should be ready for the final round of review. Cheers

File: articles/clustering-in-unsupervised-ml/index.md (hunk 1 above)
Comment (mbaabuones) [reply articles/clustering-in-unsupervised-ml/index.md]: Done.

File: articles/clustering-in-unsupervised-ml/index.md (hunk 2 above)
Comment (mbaabuones) [reply articles/clustering-in-unsupervised-ml/index.md]: I have made changes to simplify the sentence. I have also removed the self in self-discovery.

File: articles/clustering-in-unsupervised-ml/index.md (hunk 3 above)
Comment (mbaabuones) [reply articles/clustering-in-unsupervised-ml/index.md]: Done.

File: articles/clustering-in-unsupervised-ml/index.md (hunk 4 above)
Comment (mbaabuones) [reply articles/clustering-in-unsupervised-ml/index.md]: Done.

File: articles/clustering-in-unsupervised-ml/index.md (hunk 5 above)
Comment (mbaabuones) [reply articles/clustering-in-unsupervised-ml/index.md:113]: I have created a new section titled 'key concepts'. In this section, i have explained the main terms that have been used in the algorithm steps. These include MinPts, core point, border point and noise point.

File: articles/clustering-in-unsupervised-ml/index.md (hunk 6 above)
Comment (mbaabuones) [reply articles/clustering-in-unsupervised-ml/index.md]: I meant rectifying the covariance between the points artificially. I have made changes to it.

**Comment by hectorkambow** [comment]
Plagiarism check done
